apps/inviwodome/src/sgctmanager.cpp

```diff
@@ -0,0 +1,238 @@
+
+#include <inviwo/core/util/rendercontext.h>
+#include <inviwo/core/util/exception.h>
+#include <inviwo/core/util/zip.h>
+#include <inviwo/core/util/stdextensions.h>
+
+#include <inviwo/core/interaction/events/event.h>
+#include <inviwo/core/interaction/events/keyboardevent.h>
+#include <inviwo/core/interaction/events/eventpropagator.h>
+
+#include <inviwo/core/processors/processor.h>
+#include <inviwo/core/processors/canvasprocessor.h>
+#include <inviwo/core/processors/processorwidget.h>
+
+#include <inviwo/core/properties/cameraproperty.h>
+#include <inviwo/core/network/networkvisitor.h>
+#include <inviwo/core/network/lambdanetworkvisitor.h>
+#include <inviwo/core/network/processornetwork.h>
+#include <inviwo/core/network/networkutils.h>
+
+#include <modules/opengl/canvasgl.h>
+#include <modules/opengl/texture/textureunit.h>
+#include <modules/opengl/texture/textureutils.h>
+#include <modules/opengl/shader/shader.h>
+#include <modules/opengl/image/layergl.h>
+
+#include <modules/opengl/shader/shader.h>
+#include <modules/opengl/shader/shaderresource.h>
+#include <modules/opengl/shader/shadertype.h>
+#include <modules/opengl/shader/shaderobject.h>
+
+#include <modules/glfw/glfwuserdata.h>
+#include <modules/glfw/glfwwindoweventmanager.h>
+
+#include <inviwo/sgct/datastructures/sgctcamera.h>
+
+#include <sgct/engine.h>
+
+#include <inviwo/tracy/tracy.h>
+#include <inviwo/tracy/tracyopengl.h>
+
+#include <functional>
+#include <algorithm>
+#include <string_view>
+#include <algorithm>
+
+namespace inviwo {
+
+namespace {
+
+constexpr std::string_view copyVertStr = R"(
+out vec3 texCoord;
+void main() {
+    texCoord = in_TexCoord;
+    gl_Position = in_Vertex;
+}
+)";
+
+constexpr std::string_view copyFragStr = R"(
+uniform sampler2D color;
+uniform sampler2D depth;
+in vec3 texCoord;
+void main() {
+    FragData0 = texture(color, texCoord.xy);
+    gl_FragDepth = texture(depth, texCoord.xy).r;
+}
+)";
+
+#include <inviwo/core/network/networkvisitor.h>
+
+}  // namespace
+
+SGCTManager::SGCTManager(InviwoApplication& app) : app{app} {
+    app.getProcessorNetwork()->addObserver(this);
+}
+
+SGCTManager::~SGCTManager() {
+    app.getProcessorNetwork()->removeObserver(this);
+    teardownInteraction();
+}
+
+void SGCTManager::onProcessorNetworkDidAddProcessor(Processor* processor) {
+    if (!processor) return;
+
+    if (auto cp = dynamic_cast<CanvasProcessor*>(processor)) {
+        if (canvases.empty()) {
+            eventTarget = cp;
+            cp->setEvaluateWhenHidden(true);
+            if (auto widget = cp->getProcessorWidget()) {
+                widget->setVisible(false);
+            }
+        }
+        canvases.push_back(cp);
+    }
+    for (auto* camera : processor->getPropertiesByType<CameraProperty>(true)) {
+        if (cameras.empty()) {
+            sgct::Engine::instance().setNearAndFarClippingPlanes(camera->getNearPlaneDist(),
+                                                                 camera->getFarPlaneDist());
```

Comment:
Could backfire quite easily to only consider the very first camera in the network. Some processors have "dummy" cameras like the orientation indicator.
Or does this assume that all cameras are linked?

Reply:
ping

Reply:
Have not been a issue so far... might be able to traverse the network to find an appropriate camera